src/node_binding.cc

```diff
@@ -463,6 +463,10 @@ void DLOpen(const FunctionCallbackInfo<Value>& args) {
     }
 
     if (mp != nullptr) {
+      if (mp->nm_context_register_func == nullptr) {
+        if (env->options()->force_context_aware)
+          THROW_ERR_NON_CONTEXT_AWARE_DISABLED(env);
```

Comment:
Just noticing, I imagine you also want to `dlib->Close()` and `return false;` here?